### ISSUE TITLE
Reuse existing cUrl connections

### ISSUE THREAD

**Issue by x25** [pusher/pusher-http-php#36]
Do not close the cURL handler after request, so that it could reuse the existing connection. 

Example:

```php
$ch = curl_init();

curl_setopt($ch, CURLOPT_URL,"http://api.pusherapp.com");
curl_setopt($ch, CURLOPT_VERBOSE, true);

curl_exec ($ch);
curl_exec ($ch);
```

Output:

```
* Rebuilt URL to: http://api.pusherapp.com/
* Hostname was NOT found in DNS cache
*   Trying 184.73.245.186...
* Connected to api.pusherapp.com (184.73.245.186) port 80 (#0)
> GET / HTTP/1.1
Host: api.pusherapp.com
Accept: */*

< HTTP/1.1 200 OK
< Content-Type: text/html;charset=utf-8
* Server thin is not blacklisted
< Server: thin
< Content-Length: 11
< Connection: keep-alive
< 
Pusher API
* Connection #0 to host api.pusherapp.com left intact
* Found bundle for host api.pusherapp.com: 0x9f724b0
* Re-using existing connection! (#0) with host api.pusherapp.com <-- !!!
* Connected to api.pusherapp.com (184.73.245.186) port 80 (#0)
> GET / HTTP/1.1
Host: api.pusherapp.com
Accept: */*

< HTTP/1.1 200 OK
< Content-Type: text/html;charset=utf-8
* Server thin is not blacklisted
< Server: thin
< Content-Length: 11
< Connection: keep-alive
< 
Pusher API
* Connection #0 to host api.pusherapp.com left intact
```

Output for same code but using ``curl_close($ch);`` between requests:

```
* Rebuilt URL to: http://api.pusherapp.com/
* Hostname was NOT found in DNS cache
*   Trying 54.225.207.125...
* Connected to api.pusherapp.com (54.225.207.125) port 80 (#0)
> GET / HTTP/1.1
Host: api.pusherapp.com
Accept: */*

< HTTP/1.1 200 OK
< Content-Type: text/html;charset=utf-8
* Server thin is not blacklisted
< Server: thin
< Content-Length: 11
< Connection: keep-alive
< 
Pusher API
* Connection #0 to host api.pusherapp.com left intact
* Rebuilt URL to: http://api.pusherapp.com/
* Hostname was found in DNS cache
*   Trying 54.225.207.125...
* Connected to api.pusherapp.com (54.225.207.125) port 80 (#1)
> GET / HTTP/1.1
Host: api.pusherapp.com
Accept: */*

< HTTP/1.1 200 OK
< Content-Type: text/html;charset=utf-8
* Server thin is not blacklisted
< Server: thin
< Content-Length: 11
< Connection: keep-alive
< 
Pusher API
* Connection #1 to host api.pusherapp.com left intact
```

**Comment by zerrvox** [comment]
+1

**Comment by zerrvox** [comment]
Created pull request doing this #53

**Comment by leggetter** [comment]
Fixed by #53.
